README.md

```diff
@@ -1 +1,101 @@
-# edge-block-service
\ No newline at end of file
+# minibus 🚐
+
+> A edge block service to store IPFS data structures
+
+## Getting started
+
+One time set up of your cloudflare worker subdomain for dev:
+
+- `npm install` - Install the project dependencies from the monorepo root directory
+- Sign up to Cloudflare and log in with your default browser.
+- `npm i @cloudflare/wrangler -g` - Install the Cloudflare wrangler CLI
+- `wrangler login` - Authenticate your wrangler cli; it'll open your browser.
+- Copy your cloudflare account id from `wrangler whoami`
+- Update `wrangler.toml` with a new `env`. Set your env name to be the value of `whoami` on your system you can use `pnpm start` to run the worker in dev mode for you.
+
+  [**wrangler.toml**](./wrangler.toml)
+
+  ```toml
+  [env.bobbytables]
+  workers_dev = true
+  account_id = "<what does the `wrangler whoami` say>"
+  ```
+
+- Add secrets
+
+  ```sh
+    wrangler secret put SENTRY_DSN --env $(whoami) # Get from Sentry (not required for dev)
+    wrangler secret put LOGTAIL_TOKEN --env $(whoami) # Get from Logtail
+    wrangler secret put SALT --env $(whoami) # open `https://csprng.xyz/v1/api` in the browser and use the value of `Data`
+  ```
+
+- Add R2 bucket (Note that it is only available as Private Beta at the time of writing)
+
+  ```sh
+  wrangler r2 bucket create blockstore --env $(whoami)
+  # 🌀  Creating bucket "blockstore"
+  # ✨  Success!
+  ```
+
+- `npm run publish` - Publish the worker under your env. An alias for `wrangler publish --env $(whoami)`
+- `npm dev` - Run the worker in dev mode. An alias for `wrangler dev --env $(whoami)`
+
+You only need to `npm dev` for subsequent runs. PR your env config to the `wrangler.toml` to celebrate 🎉
+
+## High level architecture
+
+The Edge HTTP internal API to store IPFS data structures on R2.
+
+![High level Architecture](./minibus.web3.storage-api.jpg)
+
+## Usage
+
+You can see full HTTP API specification at https://web3.storage/block-service-docs.
+
+### 🔒 `PUT /`
+
+> Store individuals blocks by multihash in R2.
+
+TODO: curl example
+
+### 🔒 `GET /:multihash`
+
+> Multihash block read interface.
+
+The block data is retrieved and discovered by multihash, in order to enable the same data to be referred to with different codecs.
+
+```sh
+curl -X GET -H 'Authorization: Bearer ACCESS_KEY' https://minibus.web3.storage/bciqjhirzogurjzpkzpykrusrktg2gcodyhds7o4zctkhyyhtznublca
+```
```

Comment:
Auth for writes makes sense, but do we really need that for reads ?

Reply:
There are read costs in R2 (small but exist), so I guess we want this to be exposed only for private services?